### PR TITLE
mlc-llm: new package and dependency

### PR DESCRIPTION
**Pull request by alex391**

### CODE REVIEW

File: var/spack/repos/builtin/packages/apache-tvm/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on("llvm@4.0:", type="build", when="+llvm")
+
+    def cmake_args(self):
+        args = []
+        define = self.define
+        if self.spec.satisfies("+cuda"):
+            args.append(define("USE_CUDA", True))
+        if self.spec.satisfies("+llvm"):
+            args.append(define("USE_LLVM", True))
+        return args
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/apache-tvm/package.py]: You can use a literal for `args`, and avoid the conditionals using `self.define_from_variant`.

File: var/spack/repos/builtin/packages/mlc-llm/package.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    depends_on("cxx", type="build")
+    depends_on("python@3.11", type="build")
+    depends_on("apache-tvm")
+
+    depends_on("cuda@11.8:", when="+cuda")
+
+    variant(
+        "flash-infer",
+        default=False,
+        description="Use FlashInfer? (need CUDA w/ compute capability 80;86;89;90)",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mlc-llm/package.py:35]:
```suggestion
        when="+cuda",
        description="Use FlashInfer? (need CUDA w/ compute capability 80;86;89;90)",
```

File: var/spack/repos/builtin/packages/mlc-llm/package.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    )
+    conflicts("~cuda", when="+flash-infer")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mlc-llm/package.py]:
```suggestion
```

**Review by alalazo — COMMENTED** [review]
I left a few initial comments

**Comment by alex391** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/apache-tvm/package.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    license("Apache-2.0", checked_by="alex391")
+
+    version("0.16.0", sha256="55e2629c39248ef3b1ee280e34a960182bd17bea7ae0d0fa132bbdaaf5aba1ac")
+
+    variant("llvm", default=True, description="Build with llvm for CPU codegen")
+
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("cmake@3.18:", type="build")
+    depends_on("python")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/apache-tvm/package.py]: Does this link to some Python libs? Or is this type `("build", "run")`?

**Review by alalazo — APPROVED** [review]

**Pull request merged**